src/documents/views.py

```diff
@@ -329,13 +332,15 @@ def metadata(self, request, pk=None):
 
     @action(methods=["get"], detail=True)
     def suggestions(self, request, pk=None):
-        try:
-            doc = Document.objects.get(pk=pk)
-        except Document.DoesNotExist:
-            raise Http404()
+        doc = get_object_or_404(Document, pk=pk)
 
         classifier = load_classifier()
 
+        gen = parse_date_generator(doc.filename, doc.content)
```

Comment:
So this parses the filename and/or content for each request?  Could this be cached in some way?  Or stored in the DB during the initial parse (if some upper limit is set)?

Reply:
I would say almost anything is possible. The question is whether it makes sense to cache the date values or store them in the database.

The usecase for these date suggestions is in my eyes the following: 
> A new document is imported and then the settings for that document are checked and adjusted if necessary.

In my eyes, this is a one-time task, which is why I don't think caching makes sense, since it won't take effect the first time it's loaded.

I see it similarly with the persistence of the suggestions in the database. Since this is data that is probably only needed once, I would not store something like this in the database.

What might be worth further discussion is server-side caching as part of the parsing (since this already takes some time anyway). This cache could then be retrieved and cleared the first time the document is loaded. This would have two advantages in my eyes:
* The waiting time to retrieve the suggestions when editing documents is significantly reduced.
* We can load all the document's date values and apply more complex logic for the selection of the suggestions

What do you think?

Reply:
I agree that once a user has edited the date or taken one of the suggestions, it wouldn't be needed anymore, so a persistent storage of located dates doesn't make sense.

Could the response be cached in the same way the thumbnail is, using Django's `cache_control`?  I'm not really a web expert.  That could be a benefit for the classifier loading as well being cached.

My ideal would be parsing most dates during the initial consumption, but I'm not sure how best to connect the consumption to the API in the best way.